.github/workflows/tests.yml

```diff
@@ -22,19 +22,65 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  runtime:
-    name: Runtime - ${{ matrix.os }}, Python ${{ matrix.python-version }}, Runtime ${{ matrix.runtime-version }}
+  tests:
```

Comment:
Over in #279 I proposed doing something similar to this, but without a new `category` matrix item. What would you think about just running the tests as a single job, and letting xdist do the rest?

Reply:
I'd say it's reasonable, and it would save some time. Some non-trivial engineering is needed - mind if I do it in a successive PR?

Reply:
Sure, I don't think anything here makes that refactor harder.